src/deepsparse/pipeline.py

```diff
@@ -167,11 +175,29 @@ def __init__(
         self.onnx_file_path = self.setup_onnx_file_path()
         self.engine = self._initialize_engine()
 
-    def __call__(self, threadpool=None, *args, **kwargs) -> Union[BaseModel, Future]:
-        threadpool = threadpool or self._threadpool
-        if threadpool:
-            return threadpool.submit(self._run, *args, **kwargs)
-        return self._run(*args, **kwargs)
+    def __call__(self, *args, **kwargs) -> Union[BaseModel, Future]:
+        executor = kwargs.get("executor", _NOT_PRESENT_FLAG)
+
+        if executor is None:
+            # do not use threading
+            return self._run(*args, **kwargs)
```

Comment:
is this necessary given the rest of the flow? we should just be able to bottom out to the same `self._run(*args, **kwargs)` at the bottom?